### PR TITLE
Fix: [AEA-4203] - 'repeatNo' not stored in the psu-PrescriptionStatusUpdates table

### PR DESCRIPTION
**Pull request by kris-szlapa**
## Summary

🎫 [AEA-4203](https://nhsd-jira.digital.nhs.uk/browse/AEA-4203) CPSU - 'repeatNo' not stored in the psu-PrescriptionStatusUpdates table

- Routine Change

### Details

'repeatNo' not stored in the psu-PrescriptionStatusUpdates table

### CODE REVIEW

File: packages/updatePrescriptionStatus/tests/testHandler.test.ts [hunk 1]
```diff
@@ -73,6 +73,31 @@ describe("Integration tests for updatePrescriptionStatus handler", () => {
...
+
+    mockTransact.mockReturnValue(expectedItems)
+
+    const response: APIGatewayProxyResult = await handler(event, {})
+
+    expect(response.statusCode).toEqual(201)
+    expect(JSON.parse(response.body)).toEqual(responseSingleItem)
+
+    expect(mockSend).toHaveBeenCalledTimes(1)
+    expect(mockSend).toHaveBeenCalledWith(expect.objectContaining(expectedItems))
```
Comment (anthony-nhs) [inline packages/updatePrescriptionStatus/tests/testHandler.test.ts:98]: this is the same expect as in the test "when single item in request, expect a single item sent to DynamoDB" above this, so how do we know that one contains RepeatNo and one does not?

Comment (kris-szlapa) [reply packages/updatePrescriptionStatus/tests/testHandler.test.ts:100]: Checks for `RepeateNo` value have been added:
`expect(expectedItems.input.TransactItems[0].Put.Item.RepeatNo).toEqual({S: 1})`
`expect(expectedItems.input.TransactItems[0].Put.Item.RepeatNo).toEqual(undefined)`

File: packages/updatePrescriptionStatus/tests/testDatabaseClient.test.ts [hunk 2]
```diff
@@ -27,6 +27,7 @@ describe("Unit test persistDataItems", () => {
         PatientNHSNumber: "PatientNHSNumber_1",
         PharmacyODSCode: "PharmacyODSCode_1",
         PrescriptionID: "PrescriptionID_1",
+        RepeatNo: 1,
```
Comment (originalphil) [inline packages/updatePrescriptionStatus/tests/testDatabaseClient.test.ts:30]: These additions aren't needed in this test. The values are optional in the DataItem, and they don't affect the outcome of the test.

File: packages/updatePrescriptionStatus/tests/testHandler.test.ts [hunk 3]
```diff
@@ -69,6 +69,33 @@ describe("Integration tests for updatePrescriptionStatus handler", () => {
     expect(response.statusCode).toEqual(201)
     expect(JSON.parse(response.body)).toEqual(responseSingleItem)
 
+    expect(expectedItems.input.TransactItems[0].Put.Item.RepeatNo).toEqual({S: 1})
```
Comment (originalphil) [inline packages/updatePrescriptionStatus/tests/testHandler.test.ts:72]: This test shouldn't know about RepeatNo. That should all go into the new test below.

File: packages/updatePrescriptionStatus/src/updatePrescriptionStatus.ts [hunk 4]
```diff
@@ -193,12 +194,15 @@ export function buildDataItems(
...
 
+    const repeatNo = task.input?.[0]?.valueInteger
+
     const dataItem: DataItem = {
       LastModified: task.lastModified!,
       LineItemID: task.focus!.identifier!.value!.toUpperCase(),
       PatientNHSNumber: task.for!.identifier!.value!,
       PharmacyODSCode: task.owner!.identifier!.value!.toUpperCase(),
       PrescriptionID: task.basedOn![0].identifier!.value!.toUpperCase(),
+      ...(repeatNo !== undefined && {RepeatNo: repeatNo}),
```
[the remark this answers is not in the document]
Comment (kris-szlapa) [reply packages/updatePrescriptionStatus/src/updatePrescriptionStatus.ts:205]: There is:
`when input field is absent in a single item request, expect DynamoDB item without RepeatNo field`

File: packages/updatePrescriptionStatus/tests/testDatabaseClient.test.ts (hunk 2 above)
Comment (kris-szlapa) [reply packages/updatePrescriptionStatus/tests/testDatabaseClient.test.ts]: The optional values have been removed.

File: packages/updatePrescriptionStatus/tests/testUpdatePrescriptionStatus.test.ts [hunk 5]
```diff
@@ -228,4 +228,38 @@ describe("buildDataItems", () => {
     expect(dataItems[0].PrescriptionID).toEqual(prescriptionID)
     expect(dataItems[0].PharmacyODSCode).toEqual(pharmacyODSCode)
   })
+
+  it("should include RepeatNo in data item when defined", () => {
+    const task = validTask()
+    const lineItemID = v4().toUpperCase()
```
Comment (originalphil) [inline packages/updatePrescriptionStatus/tests/testUpdatePrescriptionStatus.test.ts:234]: You don't need lines 234-236 or 239-241. Those are only needed in the above test.

Comment (kris-szlapa) [reply packages/updatePrescriptionStatus/tests/testUpdatePrescriptionStatus.test.ts]: The code has been cleaned up.

File: packages/updatePrescriptionStatus/tests/utils/testUtils.ts [hunk 6]
```diff
@@ -38,6 +38,7 @@ const TASK_VALUES = [
     id: TASK_ID_0,
     status: "completed",
     businessStatus: "Dispatched",
+    repeatNo: 1,
```
Comment (originalphil) [inline packages/updatePrescriptionStatus/tests/utils/testUtils.ts:41]: There's something a bit odd going on in this file. This line can be removed without the tests failing.

File: packages/updatePrescriptionStatus/tests/utils/testUtils.ts [hunk 7]
```diff
@@ -48,6 +49,7 @@ const TASK_VALUES = [
     id: TASK_ID_1,
     status: "in-progress",
     businessStatus: "Ready to collect",
+    repeatNo: 1,
```
Comment (originalphil) [inline packages/updatePrescriptionStatus/tests/utils/testUtils.ts:52]: Removing this line makes the tests fail.

File: packages/updatePrescriptionStatus/tests/utils/testUtils.ts [hunk 8]
```diff
@@ -87,6 +89,7 @@ export function generateEntry(index: number) {
       for: {identifier: {value: values.nhsNumber, system: NHS_NUMBER_CODESYSTEM}},
       owner: {identifier: {value: values.odsCode, system: ODS_CODE_CODESYSTEM}},
       basedOn: [{identifier: {value: values.prescriptionID, system: PRESCRIPTION_ID_CODESYSTEM}}],
+      input: [{_type: {text: "Repeat Number"}, valueInteger: values.repeatNo}],
```
Comment (originalphil) [inline packages/updatePrescriptionStatus/tests/utils/testUtils.ts:92]: This line isn't needed. It also doesn't match the structure of the input used in the unit test in testUpdatePrescriptionStatus.test.ts

Comment (kris-szlapa) [reply packages/updatePrescriptionStatus/tests/utils/testUtils.ts]: The unnecessary line has been removed.

File: packages/updatePrescriptionStatus/tests/utils/testUtils.ts (hunk 7 above)
Comment (kris-szlapa) [reply packages/updatePrescriptionStatus/tests/utils/testUtils.ts:52]: Tests fail when `repeatNo` field removed from both objects in `TASK_VALUES`

File: packages/updatePrescriptionStatus/tests/utils/testUtils.ts (hunk 6 above)
Comment (kris-szlapa) [reply packages/updatePrescriptionStatus/tests/utils/testUtils.ts]: The optional field `repeatNo` has been removed.

File: packages/updatePrescriptionStatus/tests/utils/testUtils.ts (hunk 7 above)
Comment (kris-szlapa) [reply packages/updatePrescriptionStatus/tests/utils/testUtils.ts]: The optional field `repeatNo` has been removed.

**Review by originalphil — APPROVED** [review]

**Pull request merged**